src/app/rundown-view/components/rundown/rundown.component.ts

```diff
@@ -75,13 +75,13 @@ export class RundownComponent implements OnInit, OnDestroy, OnChanges {
     this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownAutoNext(this.setAutoNextStartedToTrue.bind(this)))
     this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownReset(this.setAutoNextStartedToFalse.bind(this)))
     this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownDeactivation(this.setAutoNextStartedToFalse.bind(this)))
-    this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownTake(this.scrollViewToSegment.bind(this)))
     this.rundownEventSubscriptions.push(
-      this.rundownEventObserver.subscribeToRundownSetNext(event => {
+      this.rundownEventObserver.subscribeToRundownTake(event => {
         this.setAutoNextStartedToFalse()
         this.scrollViewToSegment(event)
       })
     )
+    this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownSetNext(this.scrollViewToSegment.bind(this)))
```

Comment:
A suggestion to minimize code and avoid state patching in the component is to base the `isAutoNextStarted` value on the rundown state changes.

If `ngOnChanges` is changed to something like
```typescript
  public ngOnChanges(changes: SimpleChanges): void {
    if ('rundown' in changes) {
      this.updateMiniShelfSegments()
      this.updateMiniShelfSegmentActionMappings()
      this.setIfRundownIsActive()
      this.updateIsAutoNextStarted() // This line is added in the example
    }
  }

  private updateIsAutoNextStarted(): void {
    const onAirPart: Part | undefined = this.rundown.segments
      .find(segment => segment.isOnAir)
      ?.parts
      .find(part => part.isOnAir)

    this.isAutoNextStarted = !!onAirPart?.autoNext
  }
```
then `subscribeToEventObservers` can be reduced to
```typescript
  private subscribeToEventObserver(): void {
    this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownTake(event => this.scrollViewToSegment(event)))
    this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownSetNext(event => this.scrollViewToSegment(event)))
  }
```
and the `setIsAutoNextStartedToFalse` and `setIsAutoNextStartedToTrue` can be removed as well.

Reply:
It's done